src/adapter/src/catalog/open/builtin_item_migration.rs

```diff
@@ -365,14 +369,20 @@ async fn migrate_builtin_items_0dt(
                 .into_iter()
                 .map(|(table_key, shard_id)| ((table_key, shard_id), next_upper, -1))
                 .collect();
-            // Ignore any errors, these shards will get cleaned up in the next upgrade.
-            // It's important to use `next_upper` here. If there was another concurrent write at
-            // `next_upper`, then `updates` are no longer valid.
-            let res =
-                write_to_migration_shard(updates, next_upper, &mut write_handle, &mut since_handle)
-                    .await;
-            if let Err(e) = res {
-                error!("Unable to remove old entries from migration shard: {e:?}");
+            if !updates.is_empty() {
+                // Ignore any errors, these shards will get cleaned up in the next upgrade.
+                // It's important to use `next_upper` here. If there was another concurrent write at
+                // `next_upper`, then `updates` are no longer valid.
+                let res = write_to_migration_shard(
+                    updates,
+                    next_upper,
```

Comment:
I'm not clear on the requirements for `next_upper`. Is this write correct for both branches where `next_upper` is set above? `next_upper` is a function of `migrated_shard_updates.is_empty()`, and this branch is a function of `migration_shards_to_finalize`.